mailbox/store/src/main/java/org/apache/james/mailbox/store/search/ListeningMessageSearchIndex.java

```diff
@@ -68,45 +65,55 @@ public boolean isHandling(Event event) {
      * something relevant is received
      */
     @Override
-    public void event(Event event) throws Exception {
-        handleMailboxEvent(event,
-            sessionProvider.createSystemSession(event.getUsername()),
-            (MailboxEvent) event);
+    public Mono<Void> reactiveEvent(Event event) {
+        try {
+            return handleMailboxEvent(event,
+                sessionProvider.createSystemSession(event.getUsername()),
+                (MailboxEvent) event);
+        } catch (Exception e) {
+            return Mono.error(e);
+        }
     }
 
-    private void handleMailboxEvent(Event event, MailboxSession session, MailboxEvent mailboxEvent) throws Exception {
+    private Mono<Void> handleMailboxEvent(Event event, MailboxSession session, MailboxEvent mailboxEvent) throws Exception {
```

Comment:
why this method is throwing Exception?

Reply:
Mapper instanciation throws but is not synchronous.

Reply:
A quick analyses shows Exception is never thrown upon mapper instanciation.

I propose to remove that boiler plate annoying code in a side pull request.

Reply:
https://issues.apache.org/jira/projects/MAILBOX/issues/MAILBOX-397?filter=allissues

Reply:
See https://github.com/linagora/james-project/pull/3338

Reply:
The rebase removed the exception.